packages/theme-check-common/src/checks/valid-schema-name/index.ts

```diff
@@ -0,0 +1,40 @@
+import { parseJSON } from '../../json';
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+import { isError } from '../../utils';
+
+export const ValidSchemaName: LiquidCheckDefinition = {
+  meta: {
+    code: 'ValidSchemaName',
+    name: 'Enforce valid schema name',
+    docs: {
+      description: 'This check is aimed at ensuring a valid schema name.',
+      recommended: true,
+      url: 'https://shopify.dev/docs/storefronts/themes/tools/theme-check/checks/valid-schema-name',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async LiquidRawTag(node) {
+        if (node.name !== 'schema' || node.body.kind !== 'json') {
+          return;
+        }
+
+        const schema = parseJSON(node.body.value);
+        if (isError(schema) && schema instanceof SyntaxError) return;
+
+        if (!schema.name.startsWith('t:') && schema.name.length > 25) {
+          context.report({
+            message: 'Schema name is too long (max 25 characters)',
+            startIndex: node.blockStartPosition.end,
+            endIndex: node.blockEndPosition.start,
```

Comment:
For the error message indices, I think we should chop it down to only cover the `"name"` value area or the key/value pair. It becomes a bit intense if it highlights the entire schema.

<img width="564" alt="image" src="https://github.com/user-attachments/assets/31dc398c-7165-4301-ba79-1b1248e99b1b">

Reply:
Fair. I'd thought about it, but there's currently no tracking of the location of schema value indices (only the indices of different pieces of liquid, like where the schema ends and begins). I'll have to build that out.

Reply:
Currently working on a PR that allows us to create a `JSONDocument` within the check -- we can traverse the document and use `offset` + `length` to get valid start and end indices. I will ping you for review once it is ready and you can take a look 😄